File: 0189-rotate-array/0189-rotate-array.java
class Solution {
    public void rotate(int[] nums, int k) {
        ArrayList<Integer> list = new ArrayList<>();
        k = k%nums.length;
        for(int i=nums.length-k;i<nums.length;i++){
            list.add(nums[i]);
        }
        for(int i=0;i<nums.length-k;i++){
            list.add(nums[i]);
        }
        for(int i=0;i<nums.length;i++){
            nums[i] = list.get(i);
        }
    }
}

File: 0189-rotate-array/0189-rotate-array.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var rotate = function(nums, k) {
    let n = nums.length;
    k = k%n;
    let arr = [];
    for(let i=n-k;i<n;i++){
        arr.push(nums[i]);
    }
    for(let i=0;i<n-k;i++){
        arr.push(nums[i]);
    }
    for(let i=0;i<n;i++){
        nums[i] = arr[i];
    }
};